packages/common/src/controllers/AccountController.ts

```diff
@@ -212,38 +197,48 @@ export default class AccountController {
     await this.refreshEntitlements?.();
   };
 
-  register = async (email: string, password: string, referrer: string, consents: CustomerConsent[]) => {
-    const { config, accessModel } = useConfigStore.getState();
-
-    useAccountStore.setState({ loading: true });
-    const response = await this.accountService.register({ config, email, password, consents, referrer });
+  register = async (email: string, password: string, referrer: string, consentsValues: CustomerConsent[]) => {
+    try {
+      const response = await this.accountService.register({ email, password, consents: consentsValues, referrer });
 
-    if (response) {
-      const { user, customerConsents } = response;
-      await this.afterLogin(user, customerConsents, accessModel);
+      if (response) {
+        const { user, customerConsents } = response;
+        await this.afterLogin(user, customerConsents);
+      }
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        const errorMessage = error.message.toLowerCase();
+
+        if (errorMessage.includes('customer already exists') || errorMessage.includes('account already exists')) {
+          throw new FormValidationError({ form: [i18next.t('account:registration.user_exists')] });
+        } else if (errorMessage.includes('invalid param password')) {
+          throw new FormValidationError({ password: [i18next.t('account:registration.invalid_password')] });
+        } else {
+          // in case the endpoint fails
+          throw new FormValidationError({ password: [i18next.t('account:registration.failed_to_create')] });
+        }
+      }
     }
 
-    await this.updatePersonalShelves();
+    return;
```

Comment:
What happens when there's no response, but also no error? Is this handled downstream or should we try to refactor like login above?

Reply:
I'm actually not sure why there even is a condition for `if (response)`. All service methods should return a response or throw (which is the case for login and register). I think we can remove this condition entirely.

Reply:
Good catch btw 😄